recipes-bsp/u-boot/bootloader-versioned.bb

```diff
@@ -16,7 +16,11 @@ OMNECT_BOOTLOADER_EMBEDDED_VERSION_IMAGESIZE ?= ""
 OMNECT_BOOTLOADER_EMBEDDED_VERSION_LOCATION ?= ""
 OMNECT_BOOTLOADER_EMBEDDED_VERSION_IMAGESIZE ?= ""
 
-inherit deploy omnect_uboot_embedded_version
+# although not needed here for version calculation, nevertheless inherit class
+# omnect_bootloader_versioning to (hopefully) cause correct re-building of
+# verisioned bootlaoder artefact instead of matching an old version in SSTATE
+# cache
+inherit deploy omnect_bootloader_versioning omnect_uboot_embedded_version
```

Comment:
this will only work in kirkstone because of:
https://github.com/omnect/meta-omnect/blob/main/classes/omnect_bootloader_versioning.bbclass#L30

in scarthgap i removed, that because we have a race condition there.

other possible method is to prevent this recipe from sstate:
```
SSTATE_SKIP_CREATION = "1"
```

Reply:
imho we should add `virtual/bootloader` to `DEPENDS` that should fix the issue.

Reply:
u-boot-karo is already part of DEPENDS in line 10 via variable OMNECT_BOOTLOADER_EMBEDDED_VERSION_BBTARGET and it is apparently not sufficient.

Reply:
imho the sstate cache is invalid, when a dependency has an invalid sstate cache.
if sw-update has the updated bootloader artefact, it should be the same as used in the wic-image.
i think we look into the wrong place.

Reply:
I tried again with SKIP_SSTATE_CREATION and it yielded the same changes in tasks dependencies now as inheriting omnect_bootloader_versioning did, so I would give that a go.